xstate-chatbot/nodejs/src/machine/vitals.js

```diff
@@ -714,6 +949,95 @@ const vitalsFlow = {
             }
           }
         },
+        kidneyrelated: {
+          id: 'kidneyrelated',
+          initial: 'prompt',
+          states: {
+            prompt: {
+              onEntry: assign((context, event) => {
+                context.grammer = grammers.binaryChoice.grammer;
+                let message = dialog.get_message(messages.kidneyrelated.prompt, context.user.locale);
+                message += dialog.get_message(grammers.binaryChoice.prompt, context.user.locale);
+                dialog.sendMessage(context, message);
+              }),
+              on: {
+                USER_MESSAGE: 'process'
+              }
+            },
+            process: {
+              onEntry: assign((context, event) => {
+                context.intention = dialog.get_intention(context.grammer, event, true);
+              }),
+              always: [
+                {
+                  cond: (context) => context.grammer == dialog.INTENTION_UNKOWN,
+                  target: 'error'
+                },
+                {
+                  actions: assign((context, event) => {
+                    context.slots.vitals.symptoms.ComKidney = context.intention;
+                   
+                  }),
+                  target: '#cancerrelated'
+                }
+              ]
+            },
+            error: {
+              onEntry: assign((context, event) => {
+                dialog.sendMessage(context, dialog.get_message(dialog.global_messages.error.optionsRetry, context.user.locale), false);
+              }),
+              always: 'prompt'
+            }
+          }
+        },
+        cancerrelated: {
+          id: 'cancerrelated',
+          initial: 'prompt',
+          states: {
+            prompt: {
+              onEntry: assign((context, event) => {
+                context.grammer = grammers.binaryChoice.grammer;
+                let message = dialog.get_message(messages.cancerrelated.prompt, context.user.locale);
+                message += dialog.get_message(grammers.binaryChoice.prompt, context.user.locale);
+                dialog.sendMessage(context, message);
+              }),
+              on: {
+                USER_MESSAGE: 'process'
+              }
+            },
+            process: {
+              onEntry: assign((context, event) => {
+                context.intention = dialog.get_intention(context.grammer, event, true);
+              }),
+              always: [
+                {
+                  cond: (context) => context.grammer == dialog.INTENTION_UNKOWN,
+                  target: 'error'
+                },
+                 {
+                  cond: (context, event) => context.slots.vitals.fateh_kit_delivered == 'NO',
+                  actions: assign((context, event) => {
+                    console.log(context.slots.vitals.fateh_kit_delivered)
```

Comment:
Don't you want to store the context.intention same as the previous 2 states.
And remove the log statement.